src/language-server/builtins.ts

```diff
@@ -251,6 +256,188 @@ export function registerDefaultBuiltins(program: WhackoProgram): void {
     },
   );
 
+  addBuiltinToProgram(
+    program,
+    "types.ref",
+    ({
+      program,
+      module,
+      typeParameters,
+      node,
+      args,
+      caller,
+      getCurrentBlock,
+    }) => {
+      const [inputType, outputType] = typeParameters;
+      const [theParameter] = args;
+
+      const inputIsUsize = typesEqual(
+        inputType,
+        getIntegerType(IntegerKind.USize),
+      );
+
+      if (!isReferenceType(outputType)) {
+        // TODO: Report error diagnostic
+        reportErrorDiagnostic(
+          program,
+          module,
+          "type",
+          node,
+          `The output type of types.ref() should return a reference type.`,
+        );
+        return theInvalidValue;
+      }
+
+      if (!isRawPointerType(inputType) && !inputIsUsize) {
+        // TODO: Report error diagnostic
+        reportErrorDiagnostic(
+          program,
+          module,
+          "type",
+          node,
+          `The input typeof types.ref() should be a RawPointer or a usize.`,
+        );
+        return theInvalidValue;
+      }
+
+      const currentBlock = getCurrentBlock();
+
+      if (inputIsUsize) {
+        const runtimeValue = ensureRuntime(caller, currentBlock, theParameter);
+        const intToPtrInstruction = buildIntToPtrInstruction(
+          caller,
+          currentBlock,
+          outputType as ClassType,
+          runtimeValue,
+        );
+        return createRuntimeValue(intToPtrInstruction, outputType);
```

Comment:
I think we can omit the `outputType`.
```suggestion
        return createRuntimeValue(intToPtrInstruction);
```

Reply:
Why?

Reply:
The default is the type of the instruction.